website/pages/docs/theming/scale-out-theme.mdx

```diff
@@ -0,0 +1,519 @@
+---
+title: Scale Out Theme
+description: How to divide a bloated theme into /theme folder.
+---
+
+As your project grows in size, it is best to keep things organized. We highly
+suggest that instead of using a single `theme.js` (or `theme.ts`) file, you
+create a `/theme` folder in its place. Inside this folder, you could have a
+directory structure that looks like this:
+
+```bash
+📁 theme
+  📄 index.js  # my main theme entrypoint
+  📄 styles.js  # all my global style overrides
+  📁 foundations
+    📄 borders.js  # all my border overrides
+    📄 index.js  # export all Foundations here.
+  📁 components
+    📄 button.js  # all my button overrides
+    📄 index.js  # export all Components style here.
+```
+
+In this time, we will show a few examples written in TypeScript that follow the above structure.
+
+---
+
+## 🌐 Global Styles
+
+Create a file named `styles.ts` directly under `/theme` folder.
+You can write the Global Styles override here.
+
+```ts live=false
+// styles.ts
+
+import { Styles } from '@chakra-ui/theme-tools'
+
+const styles: Styles = {
+  global: () => ({
+    body: {
+      // These variables are defined in /foundations
+      fontWeight: 'normal',
+      color: 'black',
+      bgColor: 'white',
+      fontSizes: '16px',
+      lineHeight: 'normal',
+      // More...
+    },
+  }),
+}
+
+export default styles
+```
+
+---
+
+## 📁 Foundations
+
+<ComponentLinks
+  github={{ package: "theme/src/foundations/" }}
+/>
+
+In this folder, we define the style variables that are commonly used in all components,
+such as `colors`, `spacing`, `shadows`, and `typography`.
+All of these settings will be combined with the default theme settings.
+
+### 📐 Borders
+You can add additional borders like this:
+
+```ts live=false
+// borders.ts
+
+const borders = {
+  none: 0,
+  "16px": "16px solid",
+  "dashed": "2px dashed",
+  "double": "2px double",
+  // More...
+}
+
+export default borders
+```
+
+### 🚩 BreakPoints 
+You can overwrite the default breakpoints like this:
+
+```ts live=false
+// breakpoints.ts
+
+import { createBreakpoints } from '@chakra-ui/theme-tools'
+
+// Breakpoints for responsive design
+const breakpoints = createBreakpoints({
+  sm: '768px',
+  md: '1024px',
+  lg: '1216px',
+  xl: '1408px',
+})
+
+export default breakpoints
+
+```
+To learn more about usage of breakpoints, [see "Responsive Styles" Page](/docs/features/responsive-styles).
+
+
+###  🎨 Colors
+If you want to use additional colors, this setting is useful.
+You can also see [default colors list here](/docs/theming/theme#colors).
+
+```ts live=false
+// colors.ts
+
+export interface ColorHues {
+  50: string
+  100: string
+  200: string
+  300: string
+  400: string
+  500: string
+  600: string
+  700: string
+  800: string
+  900: string
+}
+
+export type Colors = typeof colors
```

Comment:
Please remove any additional TypeScript type when it is not clear for what it should be used.

Reply:
These types are only needed in the chakra codebase to derive the Theme and ThemeOverride types.